projectile.el

```diff
@@ -1194,12 +1194,18 @@ function is executing."
 ;; any post-processing of the files obtained via the external command.
 (defun projectile-dir-files-alien (directory)
   "Get the files for DIRECTORY using external tools."
-  (let ((vcs (projectile-project-vcs directory)))
-    (cond
-    ((eq vcs 'git)
-     (nconc (projectile-files-via-ext-command directory (projectile-get-ext-command vcs))
-            (projectile-get-sub-projects-files directory vcs)))
-    (t (projectile-files-via-ext-command directory (projectile-get-ext-command vcs))))))
+  (let ((vcs (projectile-project-vcs directory))
+        ;; getting directory name without ":" for tramp connections
+        (c_directory (car (last (split-string directory ":")))))
+
+    (mapcar (lambda (f)
```

Comment:
I haven't looked at the test failure, but I can tell you that walking over all the files and applying an extra transformation is going to have a terrible performance impact on any non-trivial project.

Reply:
Thanks!) Ok, that's was kind of hack to make file list more compact without long prefix `/home/user/<proj_root>/` for hg repos. It's seams better place for that was `projectile-hg-command` (change hg locate to hg files). By chance `hg files` also returns project file list in more compact way without long prefix.